MainModule/Server/Dependencies/Assets/Fly.lua

```diff
@@ -1,387 +1,87 @@
-		conn = runService.Stepped:Connect(function()
-			for _,v in pairs(char:GetDescendants()) do
-				if v:IsA("BasePart") then
-					v.CanCollide = false
+end)
+RunS.Heartbeat:Connect(function()
+	if Enabled and Ch.Parent and RootPart then
+		RootPart.Velocity = ZeroVector
+		RootPart.RotVelocity = ZeroVector
+		local Direction = UIS:GetFocusedTextBox()and ZeroVector or not UIS.KeyboardEnabled and Hum.MoveDirection or(ZeroVector +
+			(UIS:IsKeyDown'W'and Vector3.new(0, 0, -1)or ZeroVector) +
+			(UIS:IsKeyDown'S'and Vector3.new(0, 0, 1)or ZeroVector) +
+			(UIS:IsKeyDown'D'and Vector3.new(1, 0, 0)or ZeroVector) +
+			(UIS:IsKeyDown'A'and Vector3.new(-1, 0, 0)or ZeroVector) +
+			(UIS:IsKeyDown'Q'and Vector3.new(0, -1, 0)or ZeroVector) +
+			((UIS:IsKeyDown'E'or UIS:IsKeyDown(Enum.KeyCode.Space))and Vector3.new(0, 1, 0)or ZeroVector))
+		Direction = Direction*2*(UIS:IsKeyDown'LeftControl'and 3 or speedVal.Value)
+		CF = CF * CFrame.new(Direction)
+		local CDirection = (Mouse.Hit.Position-Camera.CFrame.Position)
+		CDirection = Camera.CFrame.Position + (CDirection.Unit * 10000)
+		CF = CFrame.new(CF.Position, not UIS.KeyboardEnabled and Camera.CFrame.Position or CDirection)
+		if noclip.Value then
+			for Indx, Obj in ipairs(Ch:GetChildren())do
+				if Obj:IsA"BasePart"then
+					Obj.CanCollide = false
 				end
 			end
-		end)
-	end
-	
-	while flying and antiLoop == antiReLoop and Check() do
-		local new = bGyro.cframe - bGyro.cframe.p + bPos.position
-		if not dir.Forward and not dir.Backward and not dir.Up and not dir.Down and not dir.Left and not dir.Right then
-			curSpeed = 1
-		else
-			if dir.Up then
-				new = new * CFrame.new(0, curSpeed, 0)
-				curSpeed = curSpeed + speedInc
-			end
-			
-			if dir.Down then
-				new = new * CFrame.new(0, -curSpeed, 0)
-				curSpeed = curSpeed + speedInc
-			end
-			
-			if dir.Forward then
-				new = new + camera.CoordinateFrame.lookVector * curSpeed
-				curSpeed = curSpeed + speedInc
-			end
-			
-			if dir.Backward then
-				new = new - camera.CoordinateFrame.lookVector * curSpeed
-				curSpeed = curSpeed + speedInc
-			end
-			
-			if dir.Left then
-				new = new * CFrame.new(-curSpeed, 0, 0)
-				curSpeed = curSpeed + speedInc
-			end
-			
-			if dir.Right then
-				new = new * CFrame.new(curSpeed, 0, 0)
-				curSpeed = curSpeed + speedInc
-			end
-		
-			if curSpeed > topSpeed then
-				curSpeed = topSpeed
-			end
 		end
-		
-		human.PlatformStand = true
-		bPos.position = new.p
-	
-		if dir.Forward then
-			bGyro.cframe = camera.CoordinateFrame*CFrame.Angles(-math.rad(curSpeed*7.5), 0, 0)
-		elseif dir.Backward then
-			bGyro.cframe = camera.CoordinateFrame*CFrame.Angles(math.rad(curSpeed*7.5), 0, 0)
-		else
-			bGyro.cframe = camera.CoordinateFrame
-		end
-		
-		runService.RenderStepped:Wait()
-	end
-	
-	Stop()
-end
-
-function Stop()
-	flying = false
-	human.PlatformStand = false
-	
-	if humChanged then
-		humChanged:Disconnect()
+		Ch:SetPrimaryPartCFrame(CF)
+		RootPart.CFrame = CF
```

Comment:
Please do not use CFrame. Use bodymovers. Bodymovers are a superior system to CFrame. They allow better flight including proper deceleration and acceleration with the dampening property. Even with lerping. 
Also Roblox discourages the use of CFrame as it is a hacky way to make things have physics.

Bodymovers allow proper flight which is smooth, looks good. CFrame produces a bootleg flight.

Also the old :flynoclip script used CFrame. But the new one uses bodymovers and I honestly think the bodymover version is better.

Also with high speeds like 200 when for example flying on a large map the flight will become quite disorienting with CFrame.

Also the CFrame flight goes too fast because it does not have any dampening and stops immediatly unlike bodymovers as @joritochip mentioned.